src/msg/simple/Accepter.cc

```diff
@@ -354,12 +392,29 @@ void *Accepter::entry()
     } else {
       ldout(msgr->cct,0) << __func__ << " no incoming connection?  sd = " << sd
 	      << " errno " << errno << " " << cpp_strerror(errno) << dendl;
-      if (++errors > 4)
-	break;
+      if (++errors > 4){
+	    if (msgr->cct->get_module_type() == CEPH_ENTITY_TYPE_OSD && hb){
+	      if (!unhealthy){
+	      	ldout(msgr->cct,0) << "accetper has error numbers  exceeded"
+	      		<< " four and reset heartbeat timeout." << dendl;
+          	msgr->cct->get_heartbeat_map()->reset_timeout(hb,0,0);
+          	unhealthy = true; 
```

Comment:
So it looks like all of our ```reset_timeout()``` commands make the OSD immediately start failing itself, and there's no way to restore functionality? And once the Accepter considers itself unhealthy, it never returns to health?

I think we need to be able to go the other direction?

Reply:
Yes, after we reset timeout, we immediately fail the osd , for there are some unexpected errors.
Any advice for better direction?

Reply:
Consider making the state of osd choppy, I'd perfer marking down straightforwardly.